### DISCUSSION TITLE
Problems running Task in VSCODE to build with g++

### DISCUSSION THREAD

**Discussion by jfraschilla** [microsoft/vscode-cpptools#10793] [Q&A]
I am trying to build my first C++ program in VSCODE. I followed the instructions shown in [Using GCC with MinGW](https://code.visualstudio.com/docs/cpp/config-mingw).  I configured a task to use g++ as shown below. When I choose Terminal-> Run Task... -> "C/C==:g++.exe build active file" I get the following error:

> Executing task: C/C++: g++.exe build active file
> 
> Starting build...
> 
> D:\msys64\mingw64\bin\g++.exe -fdiagnostics-color=always -g D:\dev\C++\vscode_helloworld_v2\helloworld.cpp -o D:\dev\C++\vscode_helloworld_v2\helloworld.exe
> 
> The system cannot find the path specified.
> 
> Build finished with error(s).
> 
> The terminal process failed to launch (exit code: -1).
> Terminal will be reused by tasks, press any key to close it.

```
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++.exe build active file",
            "command": "D:\\msys64\\mingw64\\bin\\g++.exe",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": "build",
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

If I open a command window and enter the following it does compile and creates a helloworld.exe.
> D:\msys64\mingw64\bin\g++.exe -fdiagnostics-color=always -g D:\dev\C++\vscode_helloworld_v2\helloworld.cpp -o D:\dev\C++\vscode_helloworld_v2\helloworld.exe

If I edit the tasks.json and change the type from "cppbuild" to "shell" it works and creates a helloworld.exe.

What am I doing wrong?

**Comment by sean-mcmanus** [comment]
I don't know. I'm not able to repro the issue.

**Reply by bobbrow** [reply]
Is `D:\\msys64\\mingw64\\bin` in the %PATH%?  Or maybe you could change the `"cwd"` to `"D:\\msys64\\mingw64\\bin"`?

**Reply by sean-mcmanus** [reply]
@bobbrow Yeah, I tried both PATH scenarios, and the task will fail if it's not in the PATH and cwd isn't set to the mingw path, but I could never get the "The system cannot find the path specified." part to appear.

**Reply by bobbrow** [reply]
@jfraschilla which terminal type does VS Code open for you?  Is it CMD, PowerShell, or something else?  You can also try flipping the backslashes `\\` -> `/` to see if the terminal did a weird escaping thing on you.

**Reply by jfraschilla** [reply]
Yes, D:\msys64\mingw\bin is in my path. Changing "cwd" to "D:\\msys64\\mingw64\\bin" did not work.  The following command works from the Terminal prompt "g++.exe -fdiagnostics-color=always -g D:\dev\C++\helloworld_v3\helloworld.cpp -o D:\dev\C++\helloworld_v3\helloworld.exe" which verifies that the path includes "D:\\msys64\\mingw64\\bin".  I cannot figure out why selecting "Run Build Task..." or "Run Task..." does not work from the Terminal menu items.

**Reply by jfraschilla** [reply]
@bobbrow  it is a pwsh

**Comment by jfraschilla** [comment]
@bobbrow 
![image](https://user-images.githubusercontent.com/16528365/231020756-88dc9fdd-19f6-4ec9-bee0-b809e50986ca.png)

**Reply by sean-mcmanus** [reply]
The cppbuild task uses a special terminal (https://github.com/microsoft/vscode-cpptools/blob/main/Extension/src/LanguageServer/cppBuildTaskProvider.ts#L209 ). Any idea what path it's complaining about? Do you see any errors related to this logged in the Help->Toggle Developer Tools->Console?

**Comment by jfraschilla** [comment]
@sean-mcmanus 
![image](https://user-images.githubusercontent.com/16528365/231029478-1593fdbf-b9a6-4c36-a53a-ee02b0e843b6.png)

**Reply by sean-mcmanus** [reply]
That console logging looks okay. The deprecation warning doesn't matter and the JSON.parse syntax error is not from our extension.

**Comment by jfraschilla** [comment]
Can the following in my settings.json be the problem?

```
   "terminal.integrated.profiles.windows": {
        "Windows PowerShell": {
            "path": "C:\\Users\\jfras\\AppData\\Local\\Microsoft\\WindowsApps\\Microsoft.PowerShell_8wekyb3d8bbwe\\pwsh.exe"
        },
        "Command Prompt": {
            "path": [
                "${env:windir}\\Sysnative\\cmd.exe",
                "${env:windir}\\System32\\cmd.exe"
            ],
            "args": [],
            "icon": "terminal-cmd"
        },
        "Git Bash": {
            "source": "Git Bash"
        }
    },
    "terminal.integrated.defaultProfile.windows": "Windows PowerShell",
```

**Reply by sean-mcmanus** [reply]
I don't get the error with those settings, but it wanted "Windows PowerShell" to be "PowerShell".

**Reply by jfraschilla** [reply]
Is there a reason why I am not seeing any additional information about the error?

**Reply by jfraschilla** [reply]
@sean-mcmanus I see that this may be incorrectly marked as answered.  Can you please change this status back to open.  I may have hit the wrong button when adding a reply.  I am still seeing the same problem

**Reply by jfraschilla** [reply]
@sean-mcmanus This post on stackoverflow fixed my problem:
https://stackoverflow.com/questions/62808726/the-system-cannot-find-the-path-specified-at-the-beginning-of-cmd/62808833#62808833
